object-filter/src/main/antlr3/org/infinispan/objectfilter/impl/ql/parse/QueryRenderer.g

```diff
@@ -361,8 +361,8 @@ ftClause
          { delegate.predicateFullTextRegexp($REGEXP_LITERAL.text); }
    |  ^(FT_RANGE startRange=(LSQUARE | LCURLY) lower=ftRangeBound upper=ftRangeBound endRange=(RSQUARE | RCURLY))
          { delegate.predicateFullTextRange($startRange.type == LSQUARE,
-                                     $lower.tree.getType() == ASTERISK ? null : $lower.text,
```

Comment:
If I return this change, in the code below
```java
   public void testFullTextRangeIckle() throws Exception {
      QueryFactory qf = getQueryFactory();

      Query q = qf.create("from " + getModelFactory().getTransactionTypeName() + " t where t.amount : [23 to 45]");

      List<Transaction> list = q.list();
      assertEquals(2, list.size());
   }
```

A NullPointerException is thrown.
```
java.lang.NullPointerException
	at org.infinispan.objectfilter.impl.ql.parse.QueryRenderer.ftClause(QueryRenderer.java:8657)
	at org.infinispan.objectfilter.impl.ql.parse.QueryRenderer.fullTextExpression(QueryRenderer.java:8329)
	at org.infinispan.objectfilter.impl.ql.parse.QueryRenderer.predicate(QueryRenderer.java:3285)
	at org.infinispan.objectfilter.impl.ql.parse.QueryRenderer.searchCondition(QueryRenderer.java:3007)
	at org.infinispan.objectfilter.impl.ql.parse.QueryRenderer.whereClause(QueryRenderer.java:657)
	at org.infinispan.objectfilter.impl.ql.parse.QueryRenderer.querySpec(QueryRenderer.java:510)
	at org.infinispan.objectfilter.impl.ql.parse.QueryRenderer.queryStatement(QueryRenderer.java:379)
	at org.infinispan.objectfilter.impl.ql.parse.QueryRenderer.queryStatementSet(QueryRenderer.java:292)
	at org.infinispan.objectfilter.impl.ql.parse.QueryRenderer.statement(QueryRenderer.java:220)
	at org.infinispan.objectfilter.impl.ql.QueryParser.render(QueryParser.java:88)
	at org.infinispan.objectfilter.impl.ql.QueryParser.parseQuery(QueryParser.java:70)
	at org.infinispan.objectfilter.impl.syntax.parser.IckleParser.parse(IckleParser.java:19)
	at org.infinispan.query.dsl.embedded.impl.QueryEngine.lambda$parse$1(QueryEngine.java:663)
	at org.infinispan.query.dsl.embedded.impl.QueryCache.lambda$get$0(QueryCache.java:79)
	at org.infinispan.cache.impl.TypeConverterDelegatingAdvancedCache.lambda$convertFunction$1(TypeConverterDelegatingAdvancedCache.java:101)
	at java.util.concurrent.ConcurrentMap.computeIfAbsent(ConcurrentMap.java:324)
	at org.infinispan.cache.impl.AbstractDelegatingCache.computeIfAbsent(AbstractDelegatingCache.java:343)
	at org.infinispan.cache.impl.TypeConverterDelegatingAdvancedCache.computeIfAbsent(TypeConverterDelegatingAdvancedCache.java:161)
	at org.infinispan.query.dsl.embedded.impl.QueryCache.get(QueryCache.java:79)
	at org.infinispan.query.dsl.embedded.impl.QueryEngine.parse(QueryEngine.java:663)
	at org.infinispan.query.dsl.embedded.impl.DelegatingQuery.<init>(DelegatingQuery.java:38)
	at org.infinispan.query.dsl.embedded.impl.EmbeddedQueryFactory.create(EmbeddedQueryFactory.java:24)
	at org.infinispan.query.dsl.embedded.impl.EmbeddedQueryFactory.create(EmbeddedQueryFactory.java:11)
```